README.Rmd

```diff
@@ -148,187 +71,136 @@ brew install openjdk
-details on the aggregation method, please refer to `prepare_inputs()` function
-documentation.
-
-Example data, from the use case presented in Justeau-Allaire et al. (2021)
-is included in the package:
-
-```{r, out.width="100%", fig.width = 10}
-habitat_data <- rast(
-  system.file("extdata", "habitat_hi_res.tif", package = "restoptr")
-)
-plot(habitat_data, plg=list(x="topright"))
+Next, copy the resulting Java library (.jar) file into `java` directory.
+
+```{bash, eval = FALSE}
+cp target/restopt-*.jar ../java/
 ```
 
-To instantiate a base restoration optimization problem from such an input raster,
-use the `restopt_problem()` function:
+Finally, the package can be installed with the newly compiled Java library  using the following _R_ command.
 
-```{r}
-p <- restopt_problem(
-  existing_habitat = habitat_data, 
-  aggregation_factor = 16,
-  habitat_threshold = 0.7
-) 
+```{r, eval = FALSE}
+if (!require(remotes)) install.packages("remotes")
+remotes::install_local(".")
 ```
 
-Input data and preprocessed aggregated rasters can be accessed with:
-
-- `get_original_habiata(p)` is the input, high resolution habitat raster.
-- `get_existing_habitat(p)` is the aggregated habitat raster.
-- `get_restorable_habitat(p)` is the aggregated restorable area raster.
-- `get_cell_area(p)` is the area of each planning, in number of cells
-  from the original raster. Note that this area is not necessarily the same
-  for all planning units, as if there are NA cells in the input raster some
-  planning units can partially cover NA and non-NA cells.
+## Usage
 
-We can plot the aggregated data:
+Here we will provide a short tutorial on using the _restoptr R_ package to identify priority areas for restoration. If you haven't already, please install the package (see previous section for installation instructions). We will begin by loading the package.
 
-```{r, out.width="100%", fig.width = 10}
-plot(rast(list(p$data$existing_habitat, p$data$restorable_habitat)), nc = 2,  plg=list(x="topright"))
+```{r "load_package"}
+# load package
+library(restoptr)
 ```
 
-Then, we can add constraints to this base problem. For instance, lets add a
-locked-out constraint, to restrict the number of planning units that can be
-selected for restoration. Such a constraint can be used to account for existing
-land-use practices, feasibility of restoration activities, and stakeholder 
-preferences.
+To identify priorities for restoration, we require information on the spatial location of places that do and do not currently contain suitable habitat. We will now import an example dataset to specify such information (imported as the `habitat_data` object). Specifically, this object is a spatial grid (i.e., raster layer). Each grid cell corresponds to a candidate place for restoration (termed planning unit), and their values indicate the absence or presence of habitat within each planning unit (using values of zero and one, respectively).
```

Comment:
Maybe before going into the tutorial part, we should provide a sentence to describe the context of the dataset something like:

"An example dataset has been included in the package, it corresponds to the forest cover of a protected area in New Caledonia. This dataset is the one that was used in Justeau-Allaire et al. (2021)."

Reply:
Yeah, that's an excellent idea!

Reply:
Done!